### ISSUE TITLE
Medisoft consumable application

### ISSUE THREAD

**Issue by Zupercharged** [CleverRaven/Cataclysm-DDA#36357]
**Is your feature request related to a problem? Please describe.**  
The USB with Medisoft. Yup. The question of what to do with this item is a long standing issue within the community, with many previous discussions and yet still no actual use for the damn thing, despite not being a particularly uncommon item if you know where to look.

**Describe the solution you'd like**  
The idea is to make Medisoft USBs a 1 time consumable item in some way, which can be used on an autodoc to calibrate it for a single operation, leaving an empty USB. To be clear there are a few variables to this suggestion:

- An intermediary 'catalyst' sort of item (like a laptop or nursebot or perhaps a computer skill level?) could be made necessary to use it similar to how a syringe is needed to use mutagens to their fullest.
- To maintain the value of having a high medical skill for players and NPCs, the quality of the calibration should be equivalent to a high-intermediate skill level (7-8). This is still a significant improvement in chances for unskilled players while still rewarding players who pursue high medical skills.
- It could be that a high computer skill could factor in to improve the quality of calibration from the USB allowing it to be a bit less niche as a skill by bringing an extra benefit to people who train it.

**Describe alternatives you've considered**  
Previous closed suggestions for Medisoft applications;
https://github.com/CleverRaven/Cataclysm-DDA/issues/32272
https://github.com/CleverRaven/Cataclysm-DDA/issues/9684

**Additional context**  
My particular inspiration for this is my unwilling mutant lab start character. Having attempted this run a few times now I have come to a few realisations; first is that I am very good at gradually accumulating a mountain of CBMs, second is that as the game progresses i would very much like to put a few to use and finally, as a character with both no intention of being a qualified surgeon and a strong desire to stay alone and independent, i don't stand a reasonable chance of actually doing so.

My hope is that with a feature like this CBMs become more accessible for non doctor players while retaining the risks and by making the Medisoft a single use consumable it doesn't completely invalidate the current methods of installation in the process.

**Comment by Ker-Nes** [comment]
Why consumable? Are taking the interpretation that the medisoft has some draconic level DRM? If so, there should be some way of making copies from blank USB sticks with sufficient computer skill and the appropriate tools (maybe a control laptop?).

**Comment by Zupercharged** [comment]
Consumable mainly for balance reasons; if you could just use the one stick to do all your surgeries then that's a bit too much power for something so common and it makes installing CBMs by any other method a bit of a joke.

As for lore reasons for the stick being emptied i would assume its because whatever was on there gets uploaded to the Autodoc or whatever for use. Leaving the question of _what exactly Medisoft is_? This is ambiguous enough to assume there is some explanation for it being single use;
- Maybe it _just happens_ to be calibration settings for the particular CBM you next install
- Maybe its a diagnostic program that has to be reset by something unavailable to the player
- Maybe its designed to wipe itself to protect the data after unauthorised use or something?

Regardless, even without a specific reason as to why, the program itself is ambiguous enough to the point where most people would just assume 'that's how its is', if their character even has the computer skill to even understand how odd it may be.

**Comment by arthuredmunds** [comment]
Hey, this is my first comment for C:DDA!

Firstly, consumable software doesn't seem reasonable unless it is used to craft an item where it would be a permanent component (like a control laptop, or anything where the USB device represents a permanent storage).

I think it may be good to take a step back from the Medisoft item.  If the goal is to provide a specific boost for a particular CBM calibration, it may be a good idea to create specific items that calibrate an Autodoc for a specific application.

Based on the prevalence of autodocs within the game world, its likely that there would be calibration settings for low level CBMs like the internal chronometer, the LED tattoo, and other consumer-grade CBMs.  I doubt that there would be any sort of software that could calibrate an autodoc for highly difficult CBMs like Uncanny Dodge or Wired Reflexes, highlighting the specialized nature of the install itself.  

In my opinion, calibration devices could benefit characters that have extremely low intelligence and might not be able to install a CBM with any reasonable chance of success.

**Comment by Zupercharged** [comment]
I actually very much like where you took this! Crafted calibration consumables are a good idea to make the software use more realistic and I also like the idea of limiting its use by CBM complexity rather than just a blanket buff (however I would expect its success rates to be more reliable than it would have been otherwise). Benefiting players whos characters cannot or should not be able to install their own CBMs with a limited alternative is the whole point of this suggestion and I think you have thought of a solid solution!

Being a craftable item thought should be put into the appropriate skills and levels that should be necessary to produce it. I would think computers and electronics could work, since the medical data should be provided by the Medisoft while the player must still put it all together. as for levels I think 4 and 4 would be a good start, since its not too easy to just happily grind in the first few days but also not so advanced that it gates it as late game content, by which point it would be useless.

**Comment by Zupercharged** [comment]
I was talking under the assumption that the software was the Medisoft on the USB. What the player would be crafting is the calibration module which uses the Medisoft USB as an ingredient. I don't expect anyone to be writing software, but to be making hardware to use the software in.

**Comment by Marrim** [comment]
It could be non-consumable but each copy is only usable with a particular CBM type.

**Comment by Zupercharged** [comment]
I mean, that would work to much the same effect since youd be unlikely to use it twice. Specific recipies would add a lot of clutter to the crafting menus though which would be nice to avoid if possible.

**Comment by myoia** [comment]
I'm surprised I didn't think about this sooner: this could be used to craft an inactive nurse bot.

**Comment by kevingranade** [comment]
See https://github.com/CleverRaven/Cataclysm-DDA/issues/32272 and https://github.com/CleverRaven/Cataclysm-DDA/pull/22877#issuecomment-362961224 and https://github.com/CleverRaven/Cataclysm-DDA/issues/27750#issuecomment-457048283

Tl;dr a universal piece of software that improves install chances is a big no, and the single-use mechanism doesn't make any sense.

As for the issue as a whole, there's no problem withiteks that have no use.
